addons/udes_stock/models/stock_move_line.py

```diff
@@ -654,5 +657,15 @@ def _validate_location_dest(self):
         for picking in self.mapped('picking_id'):
             if not picking.is_valid_location_dest_id(location=location):
                 raise ValidationError(
-                    _("The location '%s' is not a child of the picking destination "
-                      "location '%s'" % (location.name, picking.location_dest_id.name)))
+                    _("The location '%s' is not a child of the picking "
+                      "destination location '%s'" %
+                      (location.name, picking.location_dest_id.name)))
+            if picking.picking_type_id.u_drop_location_constraint == 'enforce':
+                mls = self.filtered(lambda ml: ml.picking_id == picking)
+                locations = picking.get_suggested_locations(mls)
```

Comment:
Due to the two decorators, `get_suggested_locations` can be invoked twice: form field selection and move line update. Three times if we decide to present the suggestions. I still assume that we don't want to store the suggested locations for the sake of such workflow.

Reply:
This will be addressed within a separate story.